### ISSUE TITLE
The method 'findRenderObject' was called on null.

### ISSUE THREAD

**Issue by florianh01** [singerdmx/flutter-quill#52]
I found another Bug in Flutter-quill.

This is the error message:

```
I/flutter (32301): ══╡ EXCEPTION CAUGHT BY SCHEDULER LIBRARY ╞═════════════════════════════════════════════════════════
I/flutter (32301): The following NoSuchMethodError was thrown during a scheduler callback:
I/flutter (32301): The method 'findRenderObject' was called on null.
I/flutter (32301): Receiver: null
I/flutter (32301): Tried calling: findRenderObject()
I/flutter (32301): 
I/flutter (32301): When the exception was thrown, this was the stack:
I/flutter (32301): #0      Object.noSuchMethod (dart:core-patch/object_patch.dart:54:5)
I/flutter (32301): #1      RawEditorState.getRenderEditor (package:flutter_quill/widgets/raw_editor.dart:1001:38)
I/flutter (32301): #2      RawEditorState._showCaretOnScreen.<anonymous closure> (package:flutter_quill/widgets/raw_editor.dart:977:50)
I/flutter (32301): #3      SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1144:15)
I/flutter (32301): #4      SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1090:9)
I/flutter (32301): #5      SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:998:5)
I/flutter (32301): #9      _invoke (dart:ui/hooks.dart:161:10)
I/flutter (32301): #10     PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:253:5)
I/flutter (32301): #11     _drawFrame (dart:ui/hooks.dart:120:31)
I/flutter (32301): (elided 3 frames from dart:async)
```

I'm not quite sure why this happens - maybe nullsafety (just a guess)? I'm on Flutter Channel beta, 2.0.0, on macOS 11.2.1 20D74 darwin-arm, locale de-DE.

**Comment by florianh01** [comment]
Steps to reproduce:

1. git clone https://github.com/florianh01/flutter_app.git 
2. run Flutter app
2. Click on FAB
3. Select a type (Big Card is with flutter_quill)
4. Click around in Stepper and TextFields and paste in some values
5. Go Back to HomeScreen
6. Click again on FAB
7. Select a type again (no matter which one)
8. Check your log & notice bad performance on device

**Comment by ArjanAswal** [comment]
Cloning and then building the app is very time consuming.
Can you post a gif instead?

**Comment by florianh01** [comment]
Well, the error actually only occurs in the form of jank.
Nevertheless, I was able to find another error during testing, which may also be related to this error:

![ezgif-3-d03cd5e13709](https://user-images.githubusercontent.com/78513002/110134145-5dc6a800-7dcd-11eb-9893-78ef6cc2f707.gif)


Please note that I had to press the Flutter-Quill Editor twice to activate the input. The input was still written into the first TextInputField.

**Comment by ArjanAswal** [comment]
Are you using stable or master branch? Try using stable branch.
Also I don't understand the need to press the editor field twice. If the cursor is present in the editor then you should be able to type as well.
Still digging through this.

**Comment by ArjanAswal** [comment]
My suspicion is that quill editor is unable to handle the change in focus node gracefully.

**Comment by florianh01** [comment]
So I switched to Flutter-Quill Stable and Flutter Version 1.22.6 stable and the "The method 'findRenderObject' was called on null"-error is gone now - so I think it has something to do with null-safety. 

But: The issue shown in the gif above still exists. So I don't think these issues are related. As you mentioned, I would say that there is a problem with focus node. I will look into this now.

**Comment by ArjanAswal** [comment]
> As you mentioned, I would say that there is a problem with focus node.

Either that or the placeholder feature is buggy.
I suggest that you use the stable branch of flutter quill and choose the commit before which the placeholder feature was added.
That might solve this issue.

**Comment by singerdmx** [comment]
Just don't set placeholder if that is the issue

**Comment by florianh01** [comment]
Update: "The method 'findRenderObject' was called on null" is also fixed in the newest master channel in git - I used the version from pub.dev.

I played around a bit, I don't think there is a problem with the placeholder. I have removed all FocusNodes from my code and it works now. I also force a FocusScopeNode.unfocus with every step. In this way, the user selects the correct text field every time he wants to input something.

I think we can close this for now. 
Tanks again - Flutter Quill will now replace Zefyr in my project for sure :)
